### PR TITLE
Fix typings for Redux v4

### PR DESCRIPTION
**Pull request by nschulzke**
Fix for this issue: https://github.com/reduxjs/redux-thunk/issues/169

The types API should remain identical. I just removed the type specifier on Dispatch in ThunkAction and updated the generics on Dispatch<> in the "redux" module to match v4.

### CODE REVIEW

**Comment by mmiszy** [comment]
The build failed. Could you also fix https://github.com/reduxjs/redux-thunk/blob/master/test/typescript.ts ?

File: test/typescript.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ store.dispatch(dispatch => {
   dispatch({type: 'FOO'});
 });
 
-store.dispatch((dispatch, getState) => {
+store.dispatch((dispatch, getState: () => {foo: string}) => {
```
Comment (mmiszy) [inline test/typescript.ts]: The type for `getState` here should not be needed.
Does it not work without it?

**Comment by nschulzke** [comment]
It looks like now there are dependency issues. The tests pass on my computer, but are failing to build online. Not sure what to do about it, it looks like the new version of typescript-definition-tester has some ES6 code that breaks the build online.

File: test/typescript.ts (hunk 1 above)
Comment (nschulzke) [reply test/typescript.ts]: In Redux v4, Dispatch doesn't take any state object as a generic type. So as far as I can tell, there's no way to get the Store's State type into the Thunk arguments.

Comment (mmiszy) [reply test/typescript.ts]: Oh, okay, I pulled it and now I see the problem.
Yeah, I'll stick to my overrides for now :( Thank you for your time.

My current code:

```typescript
export type AppStore = Store<AppState> & {
  dispatch<R>(asyncAction: AsyncAction<R>): R;
};

export type AsyncAction<R = any> = (dispatch: AppStore['dispatch'], getState: AppStore['getState']) => R;

// store.js
const store = createStore(…) as AppStore; // :(
```

**Comment by tharakawj** [comment]
There is another PR [here](https://github.com/reduxjs/redux-thunk/pull/180) to make typing compatible for redux 4. Not sure both are trying to do the same.

**Comment by timdorr** [comment]
Went with #180. Thanks!

**Comment by timc13** [comment]
when is an npm release coming?

**Comment by TizioFittizio** [comment]
Any news on this?

**Comment by nschulzke** [comment]
I just re-wrote it according to my needs and used my own typings (different from this pull request). This library is almost unnecessary, what it does is so simple. Just take a look at the source at write your own.

**Comment by timdorr** [comment]
@TizioFittizio https://github.com/reduxjs/redux-thunk/releases/tag/v2.3.0

**Comment by apieceofbart** [comment]
shouldn't this be a major release?

**Comment by mmiszy** [comment]
IMHO it's just a bugfix.

**Pull request closed without merging**